### PR TITLE
Fix typo and add tests for fonts install endpoint

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
- Fix a typo in code: See @anton-vlasenko comment here: https://github.com/WordPress/gutenberg/pull/52704#discussion_r1290722815
- Add tests to specify when the endpoint should fail with 400 (bad request) error.

## Testing Instructions
Run WP_REST_Fonts_Library_Controller_Test PHP unit tests

### CODE REVIEW

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 1]
```diff
@@ -340,4 +340,142 @@ public function data_install_and_uninstall_fonts() {
...
+			'without slug'                    => array(
+				'font_families' => array(
+					array(
+						'fontFamily' => 'Piazzolla',
+						'name'       => 'Piazzolla',
+					),
+				),
+			),
+
+			'with improper font face propety' => array(
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
			'with improper font face property' => array(
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 2]
```diff
@@ -340,4 +340,142 @@ public function data_install_and_uninstall_fonts() {
...
+	 * @param array $files             Font files to install.
+	 */
+	public function test_install_with_improper_inputs( $font_families, $files = array() ) {
+		wp_set_current_user( self::$admin_id );
+		$install_request    = new WP_REST_Request( 'POST', '/wp/v2/fonts' );
+		$font_families_json = json_encode( $font_families );
+		$install_request->set_param( 'fontFamilies', $font_families_json );
+		$install_request->set_file_params( $files );
+		$response = rest_get_server()->dispatch( $install_request );
+		$this->assertEquals( 400, $response->get_status(), 'Response status is not 400 when font face has both donwload_from_url and uploaded_file properties.' );
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
		$this->assertSame( 400, $response->get_status(), 'Response status is not 400 when font face has both donwload_from_url and uploaded_file properties.' );
```

There's an Core initiative to replace all `assertEquals()` with `assertSame()`.

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 3]
```diff
@@ -340,4 +340,142 @@ public function data_install_and_uninstall_fonts() {
...
+							),
+						),
+					),
+				),
+			),
+
+		);
+	}
+
+
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]: No empty lines here.
```suggestion
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 4]
```diff
@@ -340,4 +340,142 @@ public function data_install_and_uninstall_fonts() {
...
+	/**
+	 * Tests failure when fonfaces has improper inputs
+	 *
+	 * @covers ::install_fonts
+	 *
+	 * @dataProvider data_install_with_improper_inputs
+	 *
+	 * @param array $font_families     Font families to install in theme.json format.
+	 * @param array $files             Font files to install.
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
	 * @param array $font_families Font families to install in theme.json format.
	 * @param array $files         Font files to install.
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 5]
```diff
@@ -340,4 +340,140 @@ public function data_install_and_uninstall_fonts() {
...
+
+	/**
+	 * Data provider for test_install_with_improper_inputs
+	 */
+	public function data_install_with_improper_inputs() {
+		$temp_file_path1 = wp_tempnam( 'Piazzola1-' );
+		file_put_contents( $temp_file_path1, 'Mocking file content' );
+
+		return array(
+			'not a font families array'       => array(
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
			'not a font families array'       
 => array(
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 6]
```diff
@@ -340,4 +340,140 @@ public function data_install_and_uninstall_fonts() {
...
+				'font_families' => 'This is not an array',
+			),
+
+			'empty array'                     => array(
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
			'empty array'                     
 => array(
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 7]
```diff
@@ -340,4 +340,140 @@ public function data_install_and_uninstall_fonts() {
...
+				'font_families' => array(),
+			),
+
+			'without slug'                    => array(
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
			'without slug'                    
 => array(
```

File: phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php [hunk 8]
```diff
@@ -340,4 +340,140 @@ public function data_install_and_uninstall_fonts() {
...
+					array(
+						'fontFamily' => 'Piazzolla',
+						'name'       => 'Piazzolla',
+						'slug'       => 'piazzolla',
+						'fontFace'   => 'This is not an array',
+					),
+				),
+			),
+
+			'with empty font face propety'    => array(
```
Comment (hellofromtonya) [inline phpunit/fonts-library/class-wp-rest-fonts-library-controller-test.php]:
```suggestion
			'with empty font face property'    => array(
```

**Review by hellofromtonya — APPROVED** [review]
Assuming I didn't again mess up the formatting for wpcs/phpcs ;), the changes LGTM 👍 Once the CI jobs are all 👍 , ready for merge. Great job @anton-vlasenko and @matiasbenedetto!

**Comment by hellofromtonya** [comment]
Well boo, seems I further messed up the alignment as WPCS is not happy. Instead of using the GH suggestion tool, I'll pull this PR down and fix it. Working on it now.

**Comment by matiasbenedetto** [comment]
Thanks for the review!

**Pull request merged**